sdk/src/main/java/io/radar/sdk/util/RadarFileSystemLogBuffer.kt

```diff
@@ -0,0 +1,212 @@
+            val radarLog = RadarLog(level, message.replace('\n', '\t')).toJson()
+
+            // Locks file operations while writing to the currentFile
+            fileOp {
+                FileOutputStream(File(directory, currentFile), true).use { fileOutputStream ->
+                    fileOutputStream.write("$radarLog\n".toByteArray())
+                }
+                size++
+            }
+        }
+    }
+
+    override fun getFlushableLogsStash(): Flushable<RadarLog> {
+        val (files, list) = fileOp {
+            val files = directory.listFiles()
+            val list = mutableListOf<RadarLog>()
+            // Now iterate over the old list of files
+            files?.forEach { file ->
+                FileInputStream(file).bufferedReader().forEachLine { line ->
+                    list.add(RadarLog.fromJson(JSONObject(line)))
+                }
+            }
+            // Create a new currentFile
+            currentFile = getNewFilePath()
+
+            // return files and logs
+            files to list
+        }
+        return object : Flushable<RadarLog> {
+
+            override fun get(): List<RadarLog> = list
+
+            override fun onFlush(success: Boolean) {
+                if (success) {
+                    fileOp {
+                        files?.forEach { it.delete() }
+                        size -= list.size % LOG_BUFFER_SIZE_LIMIT
+                    }
+                }
+                // else do nothing. These logs will be stored until a later time.
+            }
+        }
+    }
+
+    /**
+     * Checks if the current log file (not previous log files) is at the max buffer size
+     *
+     * @return true if the log buffer is full, otherwise false
+     */
+    private fun isFull(): Boolean {
+        return fileOp { size >= LOG_BUFFER_SIZE_LIMIT }
+    }
+
+    /**
+     * Clears oldest logs and adds a "purged" log line
+     */
+    private fun purgeInactiveFiles() {
+        fileOp {
+            val files = directory.listFiles()
+            if (files != null && files.isNotEmpty()) {
+                if (files.size > 2) {
+                    // Delete all previous files. By allowing 2 files, it allows a max number of logs to be at most
+                    // 2 x LOG_BUFFER_SIZE_LIMIT
+                    val otherFiles = files.toMutableList()
+                    for (file in otherFiles) {
+                        if (file.name == currentFile) {
+                            otherFiles.remove(file)
+                            break
+                        }
+                    }
+                    otherFiles.forEach { it.delete() }
+                }
+                // Store current file for sending later, and set currentFile to a new file
+                currentFile = getNewFilePath()
+            }
+            size = 0
+        }
+        write(Radar.RadarLogLevel.DEBUG, "------ purged oldest logs -----")
```

Comment:
This is sent to the server? (just curious)

Reply:
yes